pkg/parse/source.go

```diff
@@ -96,6 +96,9 @@ func (o *files) readConfigFiles(state *sourceState) status.Error {
 	if err != nil {
 		return status.PathWrapError(errors.Wrap(err, "listing files in the configs directory"), syncDir.OSPath())
 	}
+	if state.commit != commit {
```

Comment:
AFAICT this is still a race condition. The suggested fix was to pass the symlink rather than the real path into this function, then check if the real path changed after listing the files

Reply:
Good catch. Now passing the parser with its hydratedRoot and hydratedLink, the commit is recomputed after running listFiles then compared. Test was modified to match the behavior.